runtime/test/strategies/coalesce-recipes-test.js

```diff
@@ -0,0 +1,165 @@
+      particle P1
+        in Thing thing
+        consume root
+          must provide foo
+            handle thing
+
+      particle P2
+        in Thing thing
+        out OtherThing other
+        consume foo
+
+      particle P3
+        out Thing thing
+
+      recipe
+        slot 'id0' as slot0
+        copy 'mything' as thingHandle
+        P1
+          thing = thingHandle
+          consume root as slot0
+
+      recipe
+        use as thingHandle
+        create as otherHandle
+        P2
+          thing = thingHandle
+          other = otherHandle
+        P3
+          thing = thingHandle
+      
+      resource MyThing
+          start
+          []
+      store Store0 of Thing 'mything' in MyThing
+    `);
+
+    assert.isTrue(recipe.isResolved());
+    assert.lengthOf(recipe.particles, 3);
+    assert.lengthOf(recipe.slots, 2);
+  });
+  it('evaluates fates of handles of required slot in coalesced recipes', async () => {
+    let parseManifest = async (options) => {
+      return `
+        schema Thing
+
+        particle P1
+          in Thing thing
+          consume root
+            must provide foo
+              handle thing
+
+        particle P2
+          in Thing thing
+          ${options.outThingB ? 'out Thing outThing' : ''}
+          consume foo
+
+        recipe A
+          slot 'id0' as slot0
+          ${options.fateA} as thingHandle
+          P1
+            thing = thingHandle
+            consume root as slot0
+
+        recipe B
+          ${options.fateB ? `${options.fateB} as thingHandle` : ``}
+          P2
+            ${options.fateB ? 'thing = thingHandle' : ''}
+            ${options.outThingB ? 'outThing = thingHandle' : ''}
+      `;
+    };
+
+    await doCoalesceRecipes(await parseManifest({fateA: 'map'}));
+    await doCoalesceRecipes(await parseManifest({fateA: 'map', fateB: 'map'}));
+    await doCoalesceRecipes(await parseManifest({fateA: 'map', fateB: 'use'}));
+    await doCoalesceRecipes(await parseManifest({fateA: 'map', fateB: '?'}));
+    await doNotCoalesceRecipes(await parseManifest({fateA: 'map', fateB: 'use', outThingB: true}));
+    await doNotCoalesceRecipes(await parseManifest({fateA: 'map', fateB: 'copy'}));
+
+    await doCoalesceRecipes(await parseManifest({fateA: 'copy'}));
+    await doCoalesceRecipes(await parseManifest({fateA: 'copy', fateB: 'map'}));
+    await doCoalesceRecipes(await parseManifest({fateA: 'copy', fateB: 'copy'}));
+    await doNotCoalesceRecipes(await parseManifest({fateA: 'copy', fateB: 'create'}));
+
+    await doCoalesceRecipes(await parseManifest({fateA: 'use', fateB: 'use'}));
+    await doCoalesceRecipes(await parseManifest({fateA: 'use', fateB: 'use', outThingB: true}));
+    await doCoalesceRecipes(await parseManifest({fateA: 'use', fateB: '?'}));
+    await doCoalesceRecipes(await parseManifest({fateA: 'use', fateB: '?', outThingB: true}));
+    await doNotCoalesceRecipes(await parseManifest({fateA: 'use', fateB: 'create'}));
```

Comment:
Why this one is not ok?

recipe A wants to use a handle and recipe B wants to create it, seems like a good fit, WDYT?

Reply:
intuitively this feels wrong, because data flow and slots flow go the opposite directions. 
also it would require changing the fate of the handle in the original recipe. and it is possible that this handle is already bound to an existing handle in the arc (then we should not allow it).
I suggest, we start more conservatively to begin with, and relax the constraints later?

Reply:
ok, see what use cases we run into in the future 👍